### PR TITLE
Add handshake protocol

### PR DESCRIPTION
**Pull request by weshinsley**
~~This will need a bootstrap update.~~  no it won't

### CODE REVIEW

File: R/hello.R [hunk 1]
```diff
@@ -26,10 +26,12 @@ hipercow_hello <- function(progress = NULL, timeout = NULL, root = NULL) {
   ## TODO: here we also want to modify the template to use the
   ## fast-but-short queue; we can't do that until mrc-4801 is done
   ## though.
-  id <- task_create_expr(utils::sessionInfo(), submit = TRUE, root = root)
+  id <- task_create_expr(hipercow:::hipercow_handshake(),
+                         submit = TRUE, root = root)
```
Comment (richfitz) [inline R/hello.R:30]:
```
moo <- read_string(hipercow_file("comms/moo"))
task_create_expr({
  message(moo)
  "Moo"
}, submit = TRUE, root = root)
```


with comms/moo containing

```
 -----
Moooooo!
 ------
    \   ^__^
     \  (oo)\ ________
        (__)\         )\ /\
             ||------w|
             ||      ||
```

and two helper functions (which exist already in windows)

File: R/comms.R [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  invisible(system(paste0("paplay ", msg),
+                   ignore.stdout = TRUE,
+                   ignore.stderr = TRUE,
+                   wait = FALSE))
+}
+
+hipercow_speak <- function(msg, platform = Sys.info()["sysname"]) {
+  if (platform == "Linux") {
+    return(hipercow_speak_linux(msg))
+  }
```
Comment (richfitz) [inline R/comms.R:11]: else here

File: R/comms.R [hunk 3]
```diff
@@ -0,0 +1,27 @@
+hipercow_speak_linux <- function(msg) {
+  invisible(system(paste0("paplay ", msg),
+                   ignore.stdout = TRUE,
+                   ignore.stderr = TRUE,
+                   wait = FALSE))
+}
+
+hipercow_speak <- function(msg, platform = Sys.info()["sysname"]) {
```
Comment (richfitz) [inline R/comms.R:8]: if we don't have one already, perhaps use an `is_linux`  helper or something less gross than Sys.info()

Comment (richfitz) [reply R/comms.R:8]: see windows driver's driver.R we have:

```
is_windows <- function() {
  Sys.info()[["sysname"]] == "Windows"
}
```

File: R/comms.R [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  if (platform == "Linux") {
+    return(hipercow_speak_linux(msg))
+  }
+  try(audio::play(audio::load.wave(msg)))
+  invisible()
```
Comment (richfitz) [inline R/comms.R:13]:
```suggestion
  if (is_linux) {
    hipercow_speak_linux(msg)
  } else {
    try(audio::play(audio::load.wave(msg)))
  }
  invisible()
```

File: R/comms.R [hunk 5]
```diff
@@ -0,0 +1,27 @@
+hipercow_speak_linux <- function(msg) {
+  invisible(system(paste0("paplay ", msg),
```
Comment (richfitz) [inline R/comms.R:2]: system2 quotes better

File: R/hello.R [hunk 6]
```diff
@@ -26,10 +26,12 @@ hipercow_hello <- function(progress = NULL, timeout = NULL, root = NULL) {
...
   ok <- task_log_watch(id, timeout = timeout, progress = progress, root = root)
   result <- task_result(id, root = root)
   if (ok) {
+    hipercow_speak(system.file("comms/moo2.wav", package = "hipercow"))
```
Comment (richfitz) [inline R/hello.R:34]: Use the hipercow_file helper from the other suggestion.

Also, what if we randomly selected the moo?

File: R/hipercow.R [hunk 7]
```diff
@@ -6,6 +6,7 @@
 ##' @examples
 ##' hipercow()
 hipercow <- function() {
+  hipercow_speak(system.file("comms/moo1.wav", package = "hipercow"))
```
Comment (richfitz) [inline R/hipercow.R:9]: Or have hipercow_speak("moo1") and have that call to system file to within that?

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/hello.R (hunk 6 above)
Comment (weshinsley) [reply R/hello.R:34]: Will save random mooing for later when we have more moos

**Review by richfitz — APPROVED** [review]

**Comment by richfitz** [comment]
we might need to make the audio fraction of the handshape optional at some point in the future...

**Pull request merged**